http/src/request/guild/create_guild.rs

```diff
@@ -111,20 +213,49 @@ impl<'a> CreateGuild<'a> {
 
     /// Set the channels to create with the guild.
     ///
-    /// The maximum number of channels that can be provided is 500.
+    /// The maximum number of channels that can be provided is 500. A `Vec` of
+    /// [`GuildChannelFragment`]s must be passed. The channels will be created in the order they
+    /// are passed. If inserting [`CategoryChannelFragment`]s, all child channels must be after the
+    /// category, and any non-children [`GuildChannelFragment`]s can not be between the children.
+    /// To specify `PermissionOverwrite`s, set the `PermissionOverwriteType` to `Role`, and its
+    /// `RoleId` to the id of a role you create with [`CreateGuild#roles`]. This `Vec` will be
+    /// transformed into temporary instances of each channel type, and sent to Discord for
```

Comment:
Noting how it's handled internally probably isn't useful to know, an assumption can be made about it